<launch>
	<arg name="initialize" default="false" />
	<arg name="param_file_name" default="default_params" />
	<arg name="enable_reconfigure" default="false" />
	<arg name="name_prefix" default="color_blob_finder" />
	<arg name="name_postfix" default="" />
	<arg name="source1" default="/color_classifier/images" />
	<arg name="source2" default="/color_classifier/images" />
	
	<arg name="name" value="$(arg name_prefix)$(arg name_postfix)" />

	<node pkg="color_blob_finder" type="color_blob_finder_node" name="$(arg name)" output="screen" args="~source1:=$(arg source1) ~source2:=$(arg source2)" >
		<rosparam unless="$(arg initialize)" param="reconfigure" command="load" file="$(find color_blob_finder)/params/$(arg param_file_name).yaml" />
	</node>
	<group if="$(arg enable_reconfigure)" >
		<node pkg="dynamic_reconfigure" type="reconfigure_gui" name="$(arg name)_config" args="$(arg name)/reconfigure" />
	</group>
</launch>
